bo/preisblatt_test.go

```diff
@@ -0,0 +1,174 @@
+	then.AssertThat(s.T(), err, is.Nil())
+
+	// The following lines prepare the deserialized pricat for the equality; The capacity of the slices is different, so the slice has to be created manually
+	// Also somehow the Preisstaffel raises the assertion, therefore it is replaced with the original
+	deserializedPricat.Preispositionen = []com.Preisposition{deserializedPricat.Preispositionen[0]}
+	//deserializedPricat.Preispositionen[0].Preisstaffeln = []com.Preisstaffel{deserializedPricat.Preispositionen[0].Preisstaffeln[0]}
+	deserializedPricat.Preispositionen[0].Preisstaffeln = []com.Preisstaffel{pricat.Preispositionen[0].Preisstaffeln[0]}
+	then.AssertThat(s.T(), deserializedPricat, is.EqualTo(pricat))
+}
+
+// Test_Failed_PreisblattValidation verifies that the validators of Preisblatt work
+func (s *Suite) Test_Failed_PreisblattValidation() {
+	validate := validator.New()
+	invalidPreisblattMap := map[string][]interface{}{
+		"required": {
+			bo.Preisblatt{
+				Geschaeftsobjekt: bo.Geschaeftsobjekt{
+					BoTyp:             0,
+					VersionStruktur:   "",
+					ExterneReferenzen: nil,
+				},
+				Bezeichnung:     "",
+				Sparte:          0,
+				Preisstatus:     0,
+				Herausgeber:     bo.Marktteilnehmer{},
+				Gueltigkeit:     com.Zeitraum{},
+				Preispositionen: nil,
+			},
+		},
+		"required_without_all": {
+			bo.Preisblatt{},
+		},
+	}
+	VerfiyFailedValidations(s, validate, invalidPreisblattMap)
+}
+
+//  Test_Successful_Preisblatt_Validation verifies that a valid BO is validated without errors
+func (s *Suite) Test_Successful_Preisblatt_Validation() {
+	ad := func(adresse com.Adresse) *com.Adresse { return &adresse }
+	validate := validator.New()
+	validPricat := []bo.BusinessObject{
+		bo.Preisblatt{
+			Geschaeftsobjekt: bo.Geschaeftsobjekt{
+				BoTyp:             botyp.PREISBLATT,
+				VersionStruktur:   "1",
+				ExterneReferenzen: nil,
+			},
+			Bezeichnung: "Preisblatt",
+			Sparte:      sparte.STROM,
+			Preisstatus: preisstatus.ENDGUELTIG,
+			Herausgeber: bo.Marktteilnehmer{
+				Geschaeftspartner: bo.Geschaeftspartner{
+					Geschaeftsobjekt: bo.Geschaeftsobjekt{
+						BoTyp:             botyp.MARKTTEILNEHMER,
+						VersionStruktur:   "1",
+						ExterneReferenzen: nil,
+					},
+					Name1:                   "Testor von Test",
+					Geschaeftspartnerrollen: []geschaeftspartnerrolle.Geschaeftspartnerrolle{geschaeftspartnerrolle.MARKTPARTNER},
+					Partneradresse: ad(com.Adresse{
+						Postleitzahl: "12345",
+						Ort:          "Testhausen",
+						Strasse:      "Testallee",
+						Hausnummer:   "12",
+						Landescode:   landescode.DE,
+					}),
+				},
+				Marktrolle:       marktrolle.NB,
+				Rollencodenummer: "0815081508151",
+				Rollencodetyp:    rollencodetyp.BDEW,
+				Makoadresse:      "test@test.com",
+			},
+			Gueltigkeit: com.Zeitraum{
+				Startzeitpunkt: time.Date(2021, 12, 31, 22, 00, 00, 00, time.UTC),
+				Endzeitpunkt:   time.Date(9999, 12, 31, 23, 59, 59, 00, time.UTC),
```

Comment:
ich glaube in der praxis ist es besser das auf min-value zu setzen. oder failed es dann wegen dem gt/lt-check? dann wäre ein omitempty in der validation als erster eintrag ganz praktisch